display_list/display_list_image_filter_unittests.cc

```diff
@@ -759,6 +772,91 @@ TEST(DisplayListImageFilter, UnknownContents) {
   ASSERT_EQ(filter.skia_object().get(), sk_filter.get());
 }
 
+TEST(DisplayListImageFilter, LocalImageFilterBounds) {
+  auto filter_matrix = SkMatrix::MakeAll(2.0, 0.0, 10,  //
+                                         0.5, 3.0, 15,  //
+                                         0.0, 0.0, 1);
+  std::vector<sk_sp<SkImageFilter>> sk_filters{
+      SkImageFilters::Blur(5.0, 6.0, SkTileMode::kRepeat, nullptr),
+      SkImageFilters::ColorFilter(
+          SkColorFilters::Blend(SK_ColorRED, SkBlendMode::kSrcOver), nullptr),
+      SkImageFilters::Dilate(5.0, 10.0, nullptr),
+      SkImageFilters::MatrixTransform(filter_matrix,
+                                      ToSk(DlImageSampling::kLinear), nullptr),
+      SkImageFilters::Compose(
+          SkImageFilters::Blur(5.0, 6.0, SkTileMode::kRepeat, nullptr),
+          SkImageFilters::ColorFilter(
+              SkColorFilters::Blend(SK_ColorRED, SkBlendMode::kSrcOver),
+              nullptr))};
+
+  DlBlendColorFilter dl_color_filter(DlColor::kRed(), DlBlendMode::kSrcOver);
+  std::vector<std::shared_ptr<DlImageFilter>> dl_filters{
+      std::make_shared<DlBlurImageFilter>(5.0, 6.0, DlTileMode::kRepeat),
+      std::make_shared<DlColorFilterImageFilter>(dl_color_filter.shared()),
+      std::make_shared<DlDilateImageFilter>(5, 10),
+      std::make_shared<DlMatrixImageFilter>(filter_matrix,
+                                            DlImageSampling::kLinear),
+      std::make_shared<DlComposeImageFilter>(
+          std::make_shared<DlBlurImageFilter>(5.0, 6.0, DlTileMode::kRepeat),
+          std::make_shared<DlColorFilterImageFilter>(
+              dl_color_filter.shared()))};
+
+  auto persp = SkMatrix::I();
+  persp.setPerspY(10);
```

Comment:
I would set this to 0.001 or something small like that as 10 is a very radical perspective correction.